### PR TITLE
`--database-dsn` option returns directly an engine

### PR DESCRIPTION
**Pull request by PaulFarault**
#### Which issue(s) this PR fixes

Fixes None

#### Additional comments

- Add a callback to the `--database-dsn` option to return an engine directly
- Make the `Dao` class take an engine as an input instead of a database DSN

#### Agreements

- [X] I hereby declare this contribution to be licensed under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0).
- [X] I hereby agree to grant [TOSIT](https://www.tosit.io/) a copyright license to use my contributions.

### CODE REVIEW

File: tdp/cli/commands/test_init.py [hunk 1]
```diff
@@ -5,16 +5,17 @@
...
 
 
-def test_tdp_init(collection_path: Path, database_dsn: str, vars: Path):
+def test_tdp_init(collection_path: Path, database_engine: Engine, vars: Path):
     args = [
         "--collection-path",
         collection_path,
         "--database-dsn",
-        database_dsn,
+        database_engine,
```
Comment (rpignolet) [inline tdp/cli/commands/test_init.py:18]: I think, the arg should be a `str` no ?

Comment (PaulFarault) [reply tdp/cli/commands/test_init.py]: You're right, I mixed it with the `test_dag_order` tests. I fixed it, thanks!

**Review by rpignolet — APPROVED** [review]

**Comment by SteBaum** [comment]
The logs have become unreadable, the queries are returned in the logs and `tdp deploy --mock-deploy`does not return the message that the deployment was a success at the end (or it is hidden in the unreadable logs). I needed to do a `tdp browse` to check.

File: tdp/cli/params/database_dsn_option.py [hunk 2]
```diff
@@ -3,18 +3,35 @@
...
+
+    Args:
+        ctx: Click context.
+        param: Click parameter.
+        value: Database DSN.
+
+    Returns:
+        Engine from the DSN.
+    """
+    return create_engine(value, echo=True, future=True)
```
Comment (SteBaum) [inline tdp/cli/params/database_dsn_option.py:20]: this `echo=True` pollutes the logs printed in the terminal.

**Review by SteBaum — APPROVED** [review]

**Pull request merged**